site/public/js/simple-grading.js

```diff
@@ -242,34 +275,41 @@ function updateCheckpointCells(elems, scores, no_cookie) {
             'scores': new_scores,
         },
         (returned_data) => {
-            // Validate that the Simple Grader backend correctly saved components before updating
-            const expected_vals = JSON.stringify(Object.entries(new_scores).map(String));
-            const returned_vals = JSON.stringify(Object.entries(returned_data['data']).map(String));
-            if (expected_vals === returned_vals) {
-                elems.each((idx, elem) => {
-                    elem = $(elem);
-                    elem.animate({'border-right-width': '0px'}, 400); // animate the box
-                    elem.attr('data-score', elem.data('score'));      // update the score
-                    elem.attr('data-grader', elem.data('grader'));    // update new grader
-                    elem.find('.simple-grade-grader').text(elem.data('grader'));
-                });
-                window.socketClient.send({'type': 'update_checkpoint', 'elem': elems.attr('id').split('-')[3], 'user': parent.data('anon'), 'score': elems.data('score'), 'grader': elems.data('grader')});
+            if (returned_data.data && returned_data.data.date) {
+                const currentDate = new Date(returned_data.data.date);
+                if (!isNaN(currentDate.getTime())) {
+                    const formattedDate = `${currentDate.getFullYear()}-${padNumber(currentDate.getMonth() + 1)}-${padNumber(currentDate.getDate())} ${padNumber(currentDate.getHours())}:${padNumber(currentDate.getMinutes())}:${padNumber(currentDate.getSeconds())}`;
+                    elems.each((idx, elem) => {
+                        elem = $(elem);
+                        elem.animate({'border-right-width': '0px'}, 400);
+                        elem.attr('data-score', elem.data('score'));
+                        elem.attr('data-grader', elem.data('grader'));
+                        elem.attr('data-date', formattedDate);
+                        elem.find('.simple-grade-grader').text(elem.data('grader'));
+                        elem.find('.simple-grade-date').text(formattedDate);
+                    });
+                    window.socketClient.send({
+                        'type': 'update_checkpoint',
+                        'elem': elems.attr('id').split('-')[3],
+                        'user': parent.data('anon'),
+                        'score': elems.data('score'),
+                        'grader': elems.data('grader'),
+                        'date': formattedDate,
+                    });
+                }
+                else {
+                    console.log('Invalid date received:', returned_data.data.date);
+                }
             }
             else {
-                console.log('Save error: returned data:', returned_vals, 'does not match expected new data:', expected_vals);
+                console.log('Date not found in response:', returned_data);
```

Comment:
just wondering should we switch to more robust log system in near future? I looked up and seems like `loglevel` is commonly used as logging library for js. @williamjallen

Reply:
We don't really log much information except for in various error cases, so I don't really see a need to add a full JS logging library.